src/plugins/dashboard/public/application/listing/dashboard_unsaved_listing.tsx

```diff
@@ -0,0 +1,161 @@
+        <EuiFlexItem grow={false}>
+          <EuiIcon color="text" className="dshUnsavedListingItem__icon" type="dashboardApp" />
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiTitle size="xxs">
+            <h4 className="dshUnsavedListingItem__title">
+              {dashboard?.title ?? getNewDashboardTitle()}
+            </h4>
+          </EuiTitle>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiFlexGroup
+        gutterSize="none"
+        alignItems="flexStart"
+        className="dshUnsavedListingItem__actions"
+      >
+        <EuiFlexItem grow={false}>
+          <EuiButtonEmpty
+            flush="left"
+            size="s"
+            color="primary"
+            onClick={onOpenClick}
+            data-test-subj={`edit-unsaved-${title.split(' ').join('-')}`}
+            aria-label={dashboardUnsavedListingStrings.getEditAriaLabel(title)}
+          >
+            {dashboardUnsavedListingStrings.getEditTitle()}
+          </EuiButtonEmpty>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiButtonEmpty
+            size="s"
+            color="danger"
+            onClick={onDiscardClick}
+            data-test-subj={`discard-unsaved-${title.split(' ').join('-')}`}
+            aria-label={dashboardUnsavedListingStrings.getDiscardAriaLabel(title)}
+          >
+            {dashboardUnsavedListingStrings.getDiscardTitle()}
+          </EuiButtonEmpty>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </div>
+  );
+};
+
+export const DashboardUnsavedListing = ({ redirectTo }: { redirectTo: DashboardRedirect }) => {
+  const {
+    services: {
+      dashboardPanelStorage,
+      savedDashboards,
+      core: { overlays },
+    },
+  } = useKibana<DashboardAppServices>();
+
+  const [items, setItems] = useState<JSX.Element[]>([]);
+  const [dashboardIds, setDashboardIds] = useState<string[]>([]);
+
+  const onOpen = useCallback(
+    (id?: string) => {
+      redirectTo({ destination: 'dashboard', id, editMode: true });
+    },
+    [redirectTo]
+  );
+
+  const onDiscard = useCallback(
+    (id?: string) => {
+      confirmDiscardUnsavedChanges(overlays, () => {
+        dashboardPanelStorage.clearPanels(id);
+        setDashboardIds(dashboardPanelStorage.getDashboardIdsWithUnsavedChanges());
+      });
+    },
+    [overlays, dashboardPanelStorage]
+  );
+
+  useMount(() => {
+    setDashboardIds(dashboardPanelStorage.getDashboardIdsWithUnsavedChanges());
+  });
+
+  useEffect(() => {
+    let hasNewDashboard = false;
+    const dashPromises = dashboardIds
+      .filter((id) => {
+        if (id !== DASHBOARD_PANELS_UNSAVED_ID) {
+          return true;
+        }
+        hasNewDashboard = true;
+        return false;
+      })
+      .map((dashboardId) => savedDashboards.get(dashboardId));
+    Promise.all(dashPromises).then((dashboards: DashboardSavedObject[]) => {
+      const newItems = dashboards.map((dashboard) => (
+        <DashboardUnsavedItem
```

Comment:
nit: I'd keep all jsx outside effects (very rarely see this actually). 
this would make this effect have fewer dependencies as it wouldn't depend on `onOpen`, `onDiscard`.
IMO this would make code easier to read and less chance make mistake with dependencies in the future. 
if one of the dependencies would change too often, then all `savedDashboards.get(dashboardId)` would have to rerun each time.

Reply:
What a weird layout that was. Not sure why I implemented it that way, but it's much simpler now. Thanks for noticing this.